conans/test/functional/generators/cmake_generator_test.py

```diff
@@ -46,30 +46,26 @@ def build(self):
 
 class CMakeGeneratorTest(unittest.TestCase):
 
-    def _generator_test_helper(self, os_build, compiler, compiler_version, generator):
-        try:
-            client = TestClient()
-            client.save({"conanfile.py": CONAN_RECIPE,
-                         "CMakeLists.txt": CMAKE_RECIPE,
-                         "dummy.cpp": CPP_CONTENT,
-                         "my_profile": PROFILE.format(os_build=os_build, compiler=compiler,
-                                                      compiler_version=compiler_version)})
-            client.run("install . -p my_profile")
-            client.run("build .")
-        finally:
-            self.assertNotIn("TypeError: argument of type 'NoneType' is not iterable", client.out)
-            self.assertNotIn("ERROR:", client.out)
-            self.assertIn("Check for working CXX compiler", client.out)
-            self.assertIn('cmake -G "{}"'.format(generator), client.out)
+    def _check_build_generator(self, os_build, compiler, compiler_version, generator):
```

Comment:
much better. I've used try + finally to have control over any exception caused by undefined generator (regression test), but as any exception will be treated as error and the traceback will be printed, your solution is more convenient.

Reply:
I didn't think about the regression test but it should be well covered by other tests. In this case I think the try/except was useless as you said